### DISCUSSION TITLE
Better logic for lock transition and introduce pending time

### DISCUSSION THREAD

**Discussion by MightyMilster** [iantrich/restriction-card#33] [Ideas]
Firstly, I love this card - thank you Ian! My enhancement request may have overlooked implications for other uses - as my suggestion is based on how I use it for restricting a row switch...

As far as I can tell - there are three states of the lock **icon** - Locked, Success (Fades-out), _Vanish_ (where it has disapeared). Success + Vanish are the same - regarding the state of the restriction having been released. This; visually seems a bit ilogical to me - why have the Success and Vanish states? Success gives the impression you are heading towards a new unlocked state; whereas the entity has already been unlocked at this stage.

Would it be useful for users of your card to have Success equal a pending period to when the restriction will be lifted. This would add a small time period of safety to the card until being unrestricted. Vanish would then mean unrestricted.

The roundtrip for the process would be...
Locked >> Success-In (2sec) >> Vanish (3sec) >> Success-Out (2sec) >> Locked
- Locked & Success-In = Restricted
- Vanish & Success-Out = Unrestricted

```
pending: 2
duration: 3
```

**Comment by iantrich** [comment]
I guess I see what you're saying in terms of visuals, but from a usability standpoint, I'm not seeing the reason for it. Why would you want to wait to interact with it after you unlock it?

**Comment by MightyMilster** [comment]
As background; I moved to your card after having used @thomasloven cool solution for a long while ([toggle-lock-entity-row](https://github.com/thomasloven/lovelace-toggle-lock-entity-row/blob/master/README.md)). Sadly it's depricated now and he points to restriction-card as a way forward. I was used to the lock behaviour I have tried to describe in this post.

I suppose with all the great options that you have provided; such as entering a PIN - the user has shown clear intent to unrestrict an entity. In my case I am restricting access to a garage door which we often can't see or hear when it is operating. Accidentally opening it would be a real security breach to our home. I use a single tap to activate your card. So to answer your question...
> Why would you want to wait to interact with it after you unlock it?

...It would provide a protection against an accidental double-tap. This is the way @thomasloven solution behaved and it seemed pretty clever and intuitive to me. If you want to see it in action, you would have to use an earlier Home Assistant version prior to v0.110 when they changed effeciencies in icon loading. This change broke toggle-lock-entity-row's ability to show mdi icons.

**Comment by iantrich** [comment]
An accidental double_tap should not unlock the card (anymore) there was a bug in my action-directive that I fixed in 1.2.6, unless you have `action: double_tap` as your method of unlock. You could also set your `action` to `hold` which would IMO be even more explicit.
I'm still not convinced on the `pending` proposal as your use case was actually a bug in the code.
